### PR TITLE
Filename option for GirderFileIdToVolume

### PR DESCRIPTION
**Pull request by zachmullen**
@kotfic @cjh1 PTAL

### CODE REVIEW

File: girder_worker/docker/transforms/girder.py [hunk 1]
```diff
@@ -47,22 +47,23 @@ def transform(self, **kwargs):
 
 
 class GirderFileIdToVolume(GirderClientTransform):
-    def __init__(self, _id, volume=TemporaryVolume.default, **kwargs):
+    def __init__(self, _id, volume=TemporaryVolume.default, filename=None, **kwargs):
```
Comment (cjh1) [inline girder_worker/docker/transforms/girder.py]: Are we happy with the user having to pass the filename in when we could extract if from Girder?

Comment (zachmullen) [reply girder_worker/docker/transforms/girder.py]: I considered that as another option. Ideally both would be possible. Maybe another option `use_girder_filename`?

Comment (cjh1) [reply girder_worker/docker/transforms/girder.py]: I was thinking that if filename was ```None``` we could get it from Girder?

Comment (zachmullen) [reply girder_worker/docker/transforms/girder.py]: I'd be OK with that, though it breaks the existing behavior.

Comment (cjh1) [reply girder_worker/docker/transforms/girder.py]: I guess one issue with that approach is filename conflicts, where as the default of using the uuid is going to be pretty safe from that point of view.

Comment (zachmullen) [reply girder_worker/docker/transforms/girder.py]: Right.

Comment (cjh1) [reply girder_worker/docker/transforms/girder.py]: I guess we could get the best of both worlds with <root>/<file_id>/<filename>?

Comment (zachmullen) [reply girder_worker/docker/transforms/girder.py]: I like that plan, it also mitigates the risk of any output paths conflicting with input paths.

I'll do that extra directory as the default case, but if a filename is passed explicitly, we'll leave out the UUID. That will allow clients max control, which may be useful in odd cases like where two files need to be in the same directory (e.g. mha/mhd read side effects).

Comment (cjh1) [reply girder_worker/docker/transforms/girder.py]: Sounds good.

**Comment by zachmullen** [comment]
Due to the requested changes, there are several conflicts with #247. Since they are so closely related, I'm going to just combine the two PRs into one.

**Comment by cjh1** [comment]
@zachmullen This looks good, the tests need to be updated to reflect the path changes.

**Comment by zachmullen** [comment]
Because we are now actually using `mkdir`, the errors are mostly resulting from trying to `mkdir` in a bogus directory. My plan was to just mock out `os.mkdir` in the tests, do you have a better method?

**Comment by cjh1** [comment]
That will work or prefix the directories with /tmp.

**Comment by zachmullen** [comment]
The tests have been fixed, this is ready for final review

**Pull request merged**